src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java

```diff
@@ -141,4 +141,14 @@ private void updateTransactionWithLinks(Transaction transaction,
     private String getSubmissionUri(String transactionId, String submissionId) {
         return String.format(SUBMISSION_URI_PATTERN, transactionId, submissionId);
     }
+
+    public ResponseEntity<Object> deleteAcspApplication(String userId){
+        Optional<AcspDataDao> acspData = acspRepository.findById(userId);
+        if (acspData.isPresent()) {
+            acspRepository.deleteById(userId);
```

Comment:
I think, it need to be wrapped with try catch just in case we couldn't able to delete the record because of locking etc.